config/v1/types_cluster_monitoring.go

```diff
@@ -0,0 +1,88 @@
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:internal
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/1929
+// +openshift:file-pattern=cvoRunLevel=0000_10,operatorName=config-operator,operatorOrdering=01
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=clustermonitoring,scope=Cluster
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:annotations="description=Cluster Monitoring Operators configuration API"
+// +openshift:enable:FeatureGate=ClusterMonitoringConfig
+// ClusterMonitoring is the Schema for the Cluster Monitoring Operators API
+type ClusterMonitoring struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user configuration for the Cluster Monitoring Operator
+	// +kubebuilder:validation:Required
+	Spec ClusterMonitoringSpec `json:"spec,omitempty"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status ClusterMonitoringStatus `json:"status,omitempty"`
+}
+
+// MonitoringOperatorStatus defines the observed state of MonitoringOperator
+type ClusterMonitoringStatus struct {
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:internal
+type ClusterMonitoringList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ClusterMonitoring `json:"items"`
+}
+
+// ClusterMonitoringSpec defines the desired state of Cluster Monitoring Operator
+type ClusterMonitoringSpec struct {
+	// userDefined set the deployment mode for user-defined monitoring in addition to the default platform monitoring.
+	UserDefined UserDefinedMonitoring `json:"userDefined,omitempty"`
+}
+
+// UserDefinedMonitoring config for user-defined projects.
+type UserDefinedMonitoring struct {
+	// mode defines the different configurations of UserDefinedMonitoring
+	// Valid values are Disabled and NamespaceIsolated
+	// Disabled disables monitoring for user-defined projects. This restricts the default monitoring stack, installed in the openshift-monitoring project, to monitor only platform namespaces, which prevents any custom monitoring configurations or resources from being applied to user-defined namespaces.
+	// NamespaceIsolated enables monitoring for user-defined projects with namespace-scoped tenancy. This ensures that metrics, alerts, and monitoring data are isolated at the namespace level.
+	// +kubebuilder:validation:Enum:="Disabled";"NamespaceIsolated"
+	Mode UserDefinedMode `json:"mode,omitempty"`
```

Comment:
Is this optional or required?

Reply:
We had a looong discussion about it and I made some changes, I don't remember right now what we concluded but I though this way was correct.

Reply:
done